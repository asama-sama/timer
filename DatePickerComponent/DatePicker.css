.DatePicker {
  text-align: center;
  font-weight: bold;
  font-size: 16px;
  margin-bottom: 10px;
}

.DatePicker-Arrow {
  cursor: pointer;
}