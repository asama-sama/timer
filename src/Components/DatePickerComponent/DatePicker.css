.DatePicker {
  text-align: center;
  font-weight: bold;
  font-size: 16px;
  margin-bottom: 10px;
}

.DatePicker-Arrow {
  cursor: pointer;
}

.DatePicker-Text {
  cursor: default;
  user-select: none;
}

.DatePicker-Arrow-Left {
  float: left;
  padding-left: 40px;
}

.DatePicker-Arrow-Right {
  float: right;
  padding-right: 40px;
}