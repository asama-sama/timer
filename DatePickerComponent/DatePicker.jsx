import React, {Component} from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'semantic-ui-react';
import './DatePicker.css';
import moment from 'moment';

class DatePicker extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div styleName='DatePicker'>
        <Icon name='angle left' onClick={() => this.props.subtractDay()} />
        <span>
          {moment(this.props.date).format('ddd Do MMM YYYY')}
        </span>
        <Icon name='angle right' onClick={() => this.props.addDay()} />
      </div>);
  }
}
DatePicker.propTypes = {
  date: PropTypes.any.isRequired,
  subtractDay: PropTypes.func.isRequired,
  addDay: PropTypes.func.isRequired
};
export default DatePicker;