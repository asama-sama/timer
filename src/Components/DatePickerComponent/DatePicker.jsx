import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { Icon } from 'semantic-ui-react';
import './DatePicker.css';

const DatePicker = props => {
  const { subtractDay, date, addDay } = props;
  return (
    <div styleName='DatePicker'>
      <Icon
        name='angle left'
        styleName='DatePicker-Arrow DatePicker-Arrow-Left'
        onClick={() => subtractDay()}
      />
      <span styleName='DatePicker-Text'>
        {moment(date).format('ddd Do MMM YYYY')}
      </span>
      <Icon
        name='angle right'
        styleName='DatePicker-Arrow DatePicker-Arrow-Right'
        onClick={() => addDay()}
      />
    </div>
  );
};
DatePicker.propTypes = {
  date: PropTypes.string.isRequired,
  subtractDay: PropTypes.func.isRequired,
  addDay: PropTypes.func.isRequired
};
export default DatePicker;
