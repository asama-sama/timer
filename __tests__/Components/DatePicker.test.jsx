import React from 'react';
import renderer from 'react-test-renderer';
import DatePicker from '../../src/Components/DatePickerComponent/DatePicker';

it('renders correctly', () => {
  const tree = renderer
    .create(<DatePicker
      date=''
      subtractDay={() => {}}
      addDay={() => {}}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
