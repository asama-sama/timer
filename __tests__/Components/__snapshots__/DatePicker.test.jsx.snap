// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="src-Components-DatePickerComponent-___DatePicker__DatePicker___2L_Fn"
>
  <i
    aria-hidden="true"
    className="angle left icon src-Components-DatePickerComponent-___DatePicker__DatePicker-Arrow___2hqI4 src-Components-DatePickerComponent-___DatePicker__DatePicker-Arrow-Left___3YDcC"
    onClick={[Function]}
  />
  <span
    className="src-Components-DatePickerComponent-___DatePicker__DatePicker-Text___2HjRf"
  >
    Invalid date
  </span>
  <i
    aria-hidden="true"
    className="angle right icon src-Components-DatePickerComponent-___DatePicker__DatePicker-Arrow___2hqI4 src-Components-DatePickerComponent-___DatePicker__DatePicker-Arrow-Right___1iewH"
    onClick={[Function]}
  />
</div>
`;
